akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -171,6 +173,81 @@ final class ClusterClientSettings(
       heartbeatInterval, acceptableHeartbeatPause, bufferSize, reconnectTimeout)
 }
 
+/**
+ * Declares a super type for all events emitted by the `ClusterClient`
+ * in relation to contact points being added or removed.
+ * Events are published to the Akka event stream (system.eventStream).
+ * Given that the event bus is local only, there is no serialization
+ * of them that is required.
+ */
+sealed trait ContactPointChange {
+  val contactPoint: ActorPath
+  val clusterClient: ActorRef
+}
+
+/**
+ * Emitted to the Akka event stream when contact points have been
+ * received by the ClusterClient and a new one has been added.
+ */
+final case class ContactPointAdded(
+  override val contactPoint: ActorPath,
+  override val clusterClient: ActorRef) extends ContactPointChange
+
+/**
+ * Emitted to the Akka event stream when contact points have been
+ * received by the ClusterClient and a new one has been added.
+ */
+final case class ContactPointRemoved(
+  override val contactPoint: ActorPath,
+  override val clusterClient: ActorRef) extends ContactPointChange
+
+sealed abstract class SubscribeContactPoints
+/**
+ * A convenience method for subscribing to the event stream where
+ * it is guaranteed that a sender receives the initial state
+ * of contact points prior to any events in relation to contact
+ * point changes. This message is typically used instead of
+ * subscribers requesting the initial state and then subscribing
+ * explicitly.
+ * Note that it is the sender's responsibility to unsubscribe
+ * from the channel.
+ */
+case object SubscribeContactPoints {
+  /**
+   * Java API: get the singleton instance
+   */
+  def getInstance = this
+}
+
+sealed abstract class GetContactPoints
+/**
+ * Get the contact points known to this client. A ``ContactPoints`` message
+ * will be replied. Given that this message is intended to be used in tandem
+ * with the event bus then they are supported for use only within the same
+ * actor system as the respective client.
+ */
+case object GetContactPoints extends GetContactPoints {
+  /**
+   * Java API: get the singleton instance
+   */
+  def getInstance = this
+}
+
+/**
+ * The reply to ``GetContactPoints``.
+ *
+ * @param contactPoints The presently known list of contact points.
+ */
+final case class ContactPoints(contactPoints: Set[ActorPath]) extends NoSerializationVerificationNeeded {
```

Comment:
where does that `SetWrapper` come from? is that because of `ListSet`?
the order of these doesn't matter, so you can use ordinary `Set` (`HashSet`)